resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientResponse.java

```diff
@@ -70,12 +71,38 @@ public void setClientConfiguration(ClientConfiguration configuration)
       this.processor = configuration;
    }
 
-   @Override
-   public Object getEntity()
-   {
-      abortIfClosed();
-      return super.getEntity();
-   }
+	@Override
+	public Object getEntity() {
+		abortIfClosed();
+		Object entity = super.getEntity();
+		if (entity != null) {
+			return entity;
+		}
+		//Check if the entity was previously fully consumed
+		try {
+			InputStream inputStream = getEntityStream();

```

Comment:
Oh thanks, I did not pay attention to that.
However how can it be possible for the garbage collector to call `finalize` while another thread is calling `readFrom()` on the object being finalized ? I mean, `finalize` should be invoked on the object only if there is no reference on this object anymore. Am I missing something ?